src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -1335,9 +1339,9 @@ function assignYAxes(series: any, yAxes: IAxis[]) {
             return includes(get(axis, 'seriesIndices', []), index);
         });
 
-        if (yAxisIndex !== -1) {
-            seriesItem.yAxis = yAxisIndex;
-        }
+        // for case viewBy and stackBy have one attribute, and one measure is sliced to multiple series
+        // then 'yAxis' in other series should follow the first one
+        seriesItem.yAxis = yAxisIndex !== -1 ? yAxisIndex : series[0].yAxis;
```

Comment:
this looks like side-effect mutation of seriesItem ... please adjust you solution to avoid object mutations...